### ISSUE TITLE
HTML code in fasta description

### ISSUE THREAD

**Issue by tomas-pluskal** [wurmlab/sequenceserver#380]
Hi,

I noticed that the beta versions of 1.1.0 changed the way HTML tags are rendered in FASTA file description fields. 

In version 1.0.9, HTML entities were interpreted as HTML code, which allowed us to place things like links (<a href=...) or formatted annotations into the FASTA files that are loaded to sequenceserver. However, in version 1.1.0 all HTML tags are shown as plan text instead (I assume the < and > characters are translated to their corresponding HTML entities &lt; and &gt;). 

Is this a desired change? In my view, it actually limits the scope of sequenceserver - I thought being able to add HTML formatting to the FASTA descriptions was quite useful.

**Comment by yeban** [comment]
To minimize security risks, not interpreting any unknown HTML is the right thing to do by default. Any HTML snippet not already defined in the software's source code is unknown HTML, i.e., any user input.
So the new behavior is indeed the preferred one.

Why not use the link generation feature to create custom links instead? http://www.sequenceserver.com/doc/#plugin

**Comment by tomas-pluskal** [comment]
The link generator is nice, but having some basic formatting capabilities would be nice, too (at least translating `\n` to `<br>`). Perhaps supporting something like markdown syntax for the descriptions could be nice, too?

**Comment by tomas-pluskal** [comment]
Hi, 
I would like to return to this issue. I think having an option to add simple formatting to the sequence descriptions would be nice and useful. And using a markdown parser like kramdown (https://kramdown.gettalong.org/) this would be very easy to implement. 
What do you think?

**Comment by yeban** [comment]
I see the utility of it. Currently, adding custom links requires a bit of Ruby. But with embedded markdown, users can add them to the FASTA files using Perl, Python, bash, etc. Maybe embedded markdown can become the standard for adding custom links, while the link generator remains for automatic linking to public databases based on ID/title pattern. I think this feature should be opt-in (that is, disabled by default).

**Comment by tomas-pluskal** [comment]
I agree with the opt-in. I think it is useful not only for links, but also for highlighting stuff etc. 

I can try to code this and make a pull request.

**Comment by tomas-pluskal** [comment]
@photocyte any thoughts on this?

**Comment by photocyte** [comment]
I think it is a good idea.  Markdown formatting would support encoding of links and newlines, and the other formatting would be useful too.
